### ISSUE TITLE
Screen readers don't announce checked state change of ToolStripButton

### ISSUE THREAD

**Issue by dmitrii-drobotov** [dotnet/winforms#9137]
### .NET version

.NET 8

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

When a `ToolStripButton` with `CheckOnClick = true` is pressed (via keyboard or mouse), screen readers don't announce it. For example, this is Narrator output when focusing on a checkable `ToolStripButton` and pressing Space key: 

> toolStrip1, toolStripButton1 check box unchecked, 
> Space

ToolStripButton becomes checked, but it's not announced. On the other hand, activating a `CheckBox` is announced:

> checkBox1 check box unchecked, 
> Space
> checkBox1 check box checked, 

Other checkable controls (DataGridView cell, CheckedListBox row, TreeView node) are also announced similar to CheckBox.

### Steps to reproduce

1. Add a ToolStrip to the form
2. Add ToolStripButton to the ToolStrip, set `CheckOnClick = true`
3. Start app, focus on the ToolStripButton
4. Press space/enter or click mouse

**Comment by dmitrii-drobotov** [comment]
Looks like we just need to call `AccessibilityObject.RaiseAutomationPropertyChangedEvent(UiaCore.UIA.ToggleToggleStatePropertyId, oldValue, newValue);` when  `CheckState` is changed.

**Comment by John-Qiao** [comment]
Verified this on .NET 8.0 test pass build: 8.0.100-preview.5.23276.4, issue was fixed that: the checked and unchecked state can be announced now, test result is below:
![9137-testresult](https://github.com/dotnet/winforms/assets/45864985/675fe99c-4632-4736-9010-71336df613e4)
